File: Projects/usaco/palsquare.java
/*
ID: rombin82
LANG: JAVA
TASK: palsquare
*/

import java.io.*;
import java.util.*;

public class palsquare {
	public static void main(String[] args) throws IOException {
		Scanner in = new Scanner(new File("palsquare.in"));
		PrintWriter stdout = new PrintWriter(new File("palsquare.out"));
		Runtime.getRuntime().addShutdownHook(new Thread(() -> stdout.flush()));
		int B = in.nextInt();
		for (int i = 1; i <= 300; i++)
			if (isPalindrome(Integer.toString(i * i, B)))
				stdout.println(Integer.toString(i, B).toUpperCase() + " " + Integer.toString(i * i, B).toUpperCase());
	}
	private static boolean isPalindrome(String str) {
		return new StringBuffer(str).reverse().toString().equals(str);
	}
}

File: Projects/usaco/palsquare.kt
/*
ID: rombin82
LANG: JAVA
TASK: palsquare
*/

import java.io.*
import java.util.*

object palsquare {
    @Throws(IOException::class)
    fun main(args: Array<String>) {
        val stdin = Scanner(File("palsquare.in"))
        val stdout = PrintWriter(File("palsquare.out"))
        Runtime.getRuntime().addShutdownHook(Thread { stdout.flush() })
        val B = stdin.nextInt()
        for (i in 1..300)
            if (isPalindrome(Integer.toString(i * i, B)))
                stdout.println(Integer.toString(i, B).toUpperCase() + " " + Integer.toString(i * i, B).toUpperCase())
    }

    private fun isPalindrome(str: String): Boolean {
        return StringBuffer(str).reverse().toString().equals(str)
    }
}
